registry/repository.go

```diff
@@ -21,176 +20,123 @@ const (
+	sigManifest, err := c.getSignatureManifest(ctx, desc)
 	if err != nil {
-		return nil, err
+		return nil, ocispec.Descriptor{}, err
 	}
-	if desc.Size > maxBlobSizeLimit {
-		return nil, fmt.Errorf("signature blob too large: %d", desc.Size)
+	if len(sigManifest.Blobs) == 0 {
+		return nil, ocispec.Descriptor{}, errors.New("signature manifest missing signature envelope blob")
 	}
-	return content.FetchAll(ctx, c.Repository.Blobs(), desc)
+	sigDesc := ociDescriptorFromArtifact(sigManifest.Blobs[0])
+	if sigDesc.Size > maxBlobSizeLimit {
+		return nil, ocispec.Descriptor{}, fmt.Errorf("signature blob too large: %d", sigDesc.Size)
+	}
+	sigBlob, err := content.FetchAll(ctx, c.Repository.Blobs(), sigDesc)
+	if err != nil {
+		return nil, ocispec.Descriptor{}, err
+	}
+	return sigBlob, sigDesc, nil
 }
 
-// PutSignatureManifest creates and uploads an signature artifact linking the manifest and the signature
-func (c *RepositoryClient) PutSignatureManifest(ctx context.Context, signature []byte, signatureMediaType string, subjectManifest notation.Descriptor, annotations map[string]string) (notation.Descriptor, SignatureManifest, error) {
-	signatureDesc, err := c.uploadSignature(ctx, signature, signatureMediaType)
+// PushSignature creates and uploads an signature manifest along with its
+// linked signature envelope blob. Upon successful, PushSignature returns
+// signature envelope blob and manifest descriptors.
+func (c *repositoryClient) PushSignature(ctx context.Context, blob []byte, mediaType string, subject ocispec.Descriptor, annotations map[string]string) (blobDesc, manifestDesc ocispec.Descriptor, err error) {
+	blobDesc, err = c.uploadSignature(ctx, blob, mediaType)
 	if err != nil {
-		return notation.Descriptor{}, SignatureManifest{}, err
+		return ocispec.Descriptor{}, ocispec.Descriptor{}, err
 	}
 
-	manifestDesc, err := c.uploadSignatureManifest(ctx, artifactDescriptorFromNotation(subjectManifest), signatureDesc, annotations)
+	manifestDesc, err = c.uploadSignatureManifest(ctx, subject, blobDesc, annotations)
 	if err != nil {
-		return notation.Descriptor{}, SignatureManifest{}, err
+		return ocispec.Descriptor{}, ocispec.Descriptor{}, err
 	}
 
-	signatureManifest := SignatureManifest{
-		Blob:        notationDescriptorFromArtifact(signatureDesc),
-		Annotations: annotations,
-	}
-	return notationDescriptorFromOCI(manifestDesc), signatureManifest, nil
+	return blobDesc, manifestDesc, nil
 }
 
-func (c *RepositoryClient) getArtifactManifest(ctx context.Context, manifestDigest digest.Digest) (artifactspec.Manifest, error) {
+// getSignatureManifest returns signature manifest given signature manifest
+// descriptor
+func (c *repositoryClient) getSignatureManifest(ctx context.Context, sigManifestDesc ocispec.Descriptor) (*artifactspec.Manifest, error) {
+
 	repo := c.Repository
 	repo.ManifestMediaTypes = []string{
 		artifactspec.MediaTypeArtifactManifest,
 	}
```

Comment:
We don't need to specify the media types anymore. Instead, we should check if the media type of `sigManifestDesc` matches the media type of OCI Artifact manifest.